packages/@aws-cdk/aws-stepfunctions-tasks/lib/run-batch-job.ts

```diff
@@ -0,0 +1,304 @@
+      sfn.ServiceIntegrationPattern.SYNC
+    ];
+
+    if (!supportedPatterns.includes(this.integrationPattern)) {
+      throw new Error(
+        `Invalid Service Integration Pattern: ${this.integrationPattern} is not supported to call RunBatchJob.`
+      );
+    }
+
+    // This is reuqired since environment variables must not start with AWS_BATCH;
+    // this naming convention is reserved for variables that are set by the AWS Batch service.
+    if (props.containerOverrides?.environment) {
+      Object.keys(props.containerOverrides.environment).forEach(key => {
+        if (key.match(/^AWS_BATCH/)) {
+          throw new Error(
+            `Invalid environment variable name: ${key}. Environment variable names starting with 'AWS_BATCH' are reserved.`
+          );
+        }
+      });
+    }
+  }
+
+  public bind(_task: sfn.Task): sfn.StepFunctionsTaskConfig {
+    return {
+      resourceArn: getResourceArn(
+        'batch',
+        'submitJob',
+        this.integrationPattern
+      ),
+      policyStatements: [
+        // Resource-level access control is not supported by Batch
+        // https://docs.aws.amazon.com/step-functions/latest/dg/batch-iam.html
+        new iam.PolicyStatement({
+          resources: ['*'],
+          actions: ['batch:SubmitJob']
+        }),
+        new iam.PolicyStatement({
+          resources: [
+            `arn:aws:events:${Aws.REGION}:${Aws.ACCOUNT_ID}:rule/StepFunctionsGetEventsForBatchJobsRule`
+          ],
+          actions: [
+            'events:PutTargets',
+            'events:PutRule',
+            'events:DescribeRule'
+          ]
+        })
+      ],
+      parameters: {
+        JobDefinition: this.props.jobDefinition.jobDefinitionArn,
+        JobName: this.props.jobName,
+        JobQueue: this.props.jobQueue.jobQueueArn,
+        Parameters: this.props.payload,
+
+        ...(this.props.array && {
+          ArrayProperties: {
+            Size: this.props.array.size
+          }
+        }),
+
+        ...(this.props.containerOverrides && {
+          ContainerOverrides: {
+            Command: this.props.containerOverrides.command,
+            Environment: this.props.containerOverrides.environment
+              ? Object.entries(this.props.containerOverrides.environment).map(
+                  ([key, value]) => ({
+                    Name: key,
+                    Value: value
+                  })
+                )
+              : undefined,
+            InstanceType: this.props.containerOverrides.instanceType?.toString(),
+            Memory: this.props.containerOverrides.memory,
+            ResourceRequirements: this.props.containerOverrides.gpuCount
+              ? [
+                  {
+                    Type: 'GPU',
+                    Value: `${this.props.containerOverrides.gpuCount}`
+                  }
+                ]
+              : undefined,
+            Vcpus: this.props.containerOverrides.vcpus
+          }
+        }),
```

Comment:
This style is quite hard to read, especially around ContainerOverrides and the ternary operator mixed in. Can we switch this around to something like this - 

```ts
  // for simple properties
  ArrayProperties: this.props.array ? { Size: this.props.array.size } : undefined,

  // for a more complex setup
  ContainerOverrides: this.configureContainerOverrides(props),
  
```

Reply:
Done.